pkg/garbagecollector/announcement_handlers.go

```diff
@@ -0,0 +1,74 @@
+package garbagecollector
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/openservicemesh/osm/pkg/announcements"
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/kubernetes/events"
+)
+
+// PatchSecretHandler patches the envoy bootstrap config secrets based on the PodAdd events
+// returns a stop channel which can be used to stop the inner handler
+func PatchSecretHandler(kubeClient kubernetes.Interface) chan struct{} {
+	podAddSubscription := events.GetPubSubInstance().Subscribe(announcements.PodAdded)
+	stop := make(chan struct{})
+
+	go func() {
+		for {
+			select {
+			case <-stop:
+				return
+			case podAddedMsg := <-podAddSubscription:
+				psubMessage, castOk := podAddedMsg.(events.PubSubMessage)
+				if !castOk {
+					log.Error().Msgf("Error casting PubSubMessage: %v", psubMessage)
+					continue
+				}
+
+				// guaranteed can only be a PodAdded event
+				addedPodObj, castOk := psubMessage.NewObj.(*v1.Pod)
+				if !castOk {
+					log.Error().Msgf("Failed to cast to *v1.Pod: %v", psubMessage.OldObj)
+					continue
+				}
+
+				podUID := addedPodObj.GetUID()
+				podUUID := addedPodObj.GetLabels()[constants.EnvoyUniqueIDLabelName]
+				podName := addedPodObj.GetName()
+				namespace := addedPodObj.GetNamespace()
+				selector := fmt.Sprintf("%s=%s", constants.EnvoyUniqueIDLabelName, podUUID)
+
+				secretList, err := kubeClient.CoreV1().Secrets(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: selector})
+
+				// the secret is uniquely identified by the osm-proxy-uuid label
+				if err == nil && len(secretList.Items) == 1 {
+					secretObj := secretList.Items[0]
+					secretObj.ObjectMeta.OwnerReferences = append(secretObj.ObjectMeta.OwnerReferences, metav1.OwnerReference{
+						APIVersion: "v1",
+						Kind:       "Pod",
+						Name:       podName,
+						UID:        podUID,
+					})
+
+					if _, err = kubeClient.CoreV1().Secrets(namespace).Update(context.Background(), &secretObj, metav1.UpdateOptions{}); err != nil {
+						log.Error().Msgf("Failed to update OwnerReference for Secret %s to reference Pod %s", secretObj.GetName(), podName)
```

Comment:
We should include this `err` in the log so we know why it failed:
```suggestion
					if _, err = kubeClient.CoreV1().Secrets(namespace).Update(context.Background(), &secretObj, metav1.UpdateOptions{}); err != nil {
						log.Error().Err(err).Msgf("Failed to update OwnerReference for Secret %s to reference Pod %s", secretObj.GetName(), podName)
```

Reply:
Namespace would also be helpful in the log.